src/model.ts

```diff
@@ -0,0 +1,82 @@
+import {Random} from "@reside-ic/random";
+
+import {BaseType} from "./base";
+
+/**
+ * Constructor for a {@link DustModel}
+ *
+ * @param base The singleton base object
+ *
+ * @param pars Parameters to pass to the model
+ *
+ */
+export type DustModelConstructable = new(base: BaseType, pars: any) => DustModel;
+
+/**
+ * Information returned by an initialised model about itself
+ */
+export type DustModelInfo = DustModelInfoElement[];
+
+/**
+ * Information about a single variable within a model, typically found
+ * within {@link DustModelInfo}
+ */
+export interface DustModelInfoElement {
+    /**
+     * Dimensions of the variable.
+     * * empty vector - a scalar
+     * * length-1 vector - length of a vector
+     * * length-2 vector - number of rows and columns of a matrix
+     * * length-n vector - size of each dimension
+     *
+     * Note that as the product of an empty vector is 1,
+     * so the product of `dim` is always equal to `length`
+     */
+    dim: number[];
+
+    /**
+     * Length of the variable - if the variable represents a vector or
+     * multidimensional array, then this is the length over all
+     * dimensions. Scalars have length 1, so that the sum of all
+     * length elements within an {@link DustModelInfo} array
+     */
+    length: number;
+
+    /** Name of the variable */
+    name: string;
+}
+
+/**
+ * The interface that dust models conform to
+ *
+ */
+export interface DustModel {
+    /**
+     * Compute initial conditions
+     * @param step Step to compute initial conditionas at
+     */
+    initial(step: number): number[];
```

Comment:
Step is not really an absolutely meaningful thing, and most of the time people will initialise at step 0. However, they might not! In the covid model we used the transform of `time = step * dt`, where `dt` was typically 0.25 and time was measured as days into 2020 (so `2020-03-24` was time 84). Early on, they might start the model around Feb because that was the time of covid